### PR TITLE
Fixed allure plugin "Unexpected endStep()" error 

### PR DESCRIPTION
**Pull request by abhimanyupandian**
What triggered the error : Using actions from "I" inside the "event.dispatcher.on(event.test.failed, async (){})" within a plugin results in the following error to be displayed in the console :
```shell
allure-js-commons: Unexpected endStep(). There are no any steps running
```
Verified the fix by removing and adding the changes for the same test suite.

A similar issue (probably due to some other trigger) has been reported by others as well. See [this](https://github.com/codeceptjs/CodeceptJS/issues/1414)

## Motivation/Description of the PR
- Description of this PR, which problem it solves
- Resolves #issueId (if applicable).

Applicable helpers:

- [ ] WebDriver
- [ ] Puppeteer
- [ ] Nightmare
- [ ] REST
- [ ] FileHelper
- [ ] Appium
- [ ] Protractor
- [ ] TestCafe
- [ ] Playwright

Applicable plugins:

- [x] allure
- [ ] autoDelay
- [ ] autoLogin
- [ ] customLocator
- [ ] pauseOnFail
- [ ] coverage
- [ ] retryFailedStep
- [ ] screenshotOnFail
- [ ] selenoid
- [ ] stepByStepReport
- [ ] wdio
- [ ] subtitles

## Type of change

- [ ] :fire: Breaking changes
- [ ] :rocket: New functionality
- [x] :bug: Bug fix
- [ ] :clipboard: Documentation changes/updates
- [ ] :hotsprings: Hot fix
- [ ] :hammer: Markdown files fix - not related to source code
- [ ] :nail_care: Polish code

## Checklist:

- [ ] Tests have been added
- [ ] Documentation has been added (Run `npm run docs`)
- [ ] Lint checking (Run `npm run lint`)
- [ ] Local tests are passed (Run `npm test`)

### CODE REVIEW

File: lib/plugin/allure.js [hunk 1]
```diff
@@ -283,7 +283,10 @@ module.exports = (config) => {
   let maxLevel;
   function finishMetastep(level) {
     const metaStepsToFinish = currentMetaStep.splice(maxLevel - level);
-    metaStepsToFinish.forEach(() => reporter.endStep('passed'));
+    metaStepsToFinish.forEach(() => {
+      // only if the current step is of type Step, end it. 
+      if (reporter.suites[0].currentStep.constructor.name === 'Step') { reporter.endStep('passed'); }
```
Comment (DavertMik) [inline lib/plugin/allure.js:288]: sorry, I'm not sure that `reporter.suites[0].currentStep` will not throw exception by itself. Could you add a check that all of those objects do exist before calling them?

**Review by DavertMik — CHANGES_REQUESTED** [review]

File: lib/plugin/allure.js (hunk 1 above)
Comment (abhimanyupandian) [reply lib/plugin/allure.js]: Sure

**Comment by abhimanyupandian** [comment]
@DavertMik - Puppeteer UTs are failing since this https://github.com/codeceptjs/CodeceptJS/actions/runs/1405398832.
So unable to get the UTs passing for the current PR.

**Comment by DavertMik** [comment]
Thanks, i will check this

**Review by DavertMik — APPROVED** [review]

**Pull request merged**